sound/soc/sof/intel/hda.c

```diff
@@ -376,8 +377,12 @@ int hda_dsp_post_fw_run(struct snd_sof_dev *sdev)
 		/* Check if IMR boot is usable */
 		if (!sof_debug_check_flag(SOF_DBG_IGNORE_D3_PERSISTENT) &&
 		    (sdev->fw_ready.flags & SOF_IPC_INFO_D3_PERSISTENT ||
-		     sdev->pdata->ipc_type == SOF_IPC_TYPE_4))
+		     sdev->pdata->ipc_type == SOF_IPC_TYPE_4)) {
 			hdev->imrboot_supported = true;
+			debugfs_create_bool("request_cold_boot",
+					    0644, sdev->debugfs_root,
+					    &hdev->skip_imr_boot);
+		}
```

Comment:
The other name that came up was `request_dsp_cold_boot` to imply that it is only for the next DSP boot.
I'm fine naming it `skip_imr_boot` though.

Reply:
This option has a slight flaw and would work only to skip IMR boot for the second iteration of FW boot after the value is written isnt it?

Reply:
@ranj063, it will skip the IMR boot at the next DSP boot.
If the DSP was off then the next boot will be a 'clean' boot.
If the DSP was not off then it needs to be power cycled.